### ISSUE TITLE
05. PR1: hflip / factor_h 条件の anchor_raw_indices_view alignment test を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#424]
# 05. PR1: hflip / factor_h 条件の anchor_raw_indices_view alignment test を追加する

## 目的

`anchor_raw_indices_view` が、current view の row order に正しく追従していることを regression test で固定する。

この test は後続 PR の memory consistency loss に対する安全装置になる。ここがずれると、別 trace の過去予測を教師にしてしまう。

## 依存 issue

- `03_issue_emit_anchor_raw_indices_view.md`
- `04_issue_add_identity_contract_tests.md`

## 対象ファイル

```text
packages/seisai-engine/tests/test_fbpick_coarse.py
```

必要なら import に `project_trace_label_view` を追加する。

```python
from seisai_transforms.view_projection import (
    project_fb_idx_view,
    project_offsets_view,
    project_trace_label_view,
)
```

### 1. hflip 条件の test を追加する

既存の `_build_global_anchor_train_sample()` を使い、`augment_cfg={'hflip_prob': 1.0}` で forced hflip sample を作る。

推奨 test 名:

```python
def test_global_anchor_train_dataset_hflip_projects_anchor_raw_indices_view(
    tmp_path: Path,
) -> None:
    ...
```

検証は `project_trace_label_view()` と同じ結果になることを確認する。

```python
meta_projection = {
    'hflip': sample['meta']['hflip'],
    'factor_h': sample['meta']['factor_h'],
}
expected = project_trace_label_view(
    sample['anchor_raw_indices'].numpy(),
    H=256,
    meta=meta_projection,
)
trace_valid = np.asarray(sample['meta']['trace_valid'], dtype=np.bool_)
expected[~trace_valid] = -1

np.testing.assert_array_equal(
    sample['anchor_raw_indices_view'].numpy(),
    expected,
)
```

加えて、`sample['meta']['hflip'] is True` または truthy であることを確認する。

### 2. space.factor_h 条件の test を追加する

既存 test `test_global_anchor_train_dataset_projects_space_labels_and_offsets()` に assertion を追加してもよいし、新規 test として分けてもよい。

推奨方針は、既存 test に `anchor_raw_indices_view` の assertion を追加すること。すでに forced `factor_h=0.5` の sample を作っており、`project_fb_idx_view` と `project_offsets_view` を検証しているため、同じ `meta_projection` を使える。

追加 assertion 例:

```python
expected_anchor_raw_indices_view = project_trace_label_view(
    sample['anchor_raw_indices'].numpy(),
    H=256,
    meta=meta_projection,
)
trace_valid = np.asarray(sample['meta']['trace_valid'], dtype=np.bool_)
expected_anchor_raw_indices_view[~trace_valid] = -1

np.testing.assert_array_equal(
    sample['anchor_raw_indices_view'].numpy(),
    expected_anchor_raw_indices_view,
)
```

### 3. time-only / trace_tshift の扱いを壊さない

この issue で明示 test までは必須ではないが、実装上は以下を守る。

```text
time crop/stretch/start/factor は raw trace identity を変えない
trace_tshift_view は raw trace identity を変えない
```

`anchor_raw_indices_view` の projection は trace-axis mapping のみでよい。

## 非目標

この issue では以下を実装しない。

- memory bank / consistency loss
- confidence gate
- train config 追加
- raw inference dataset の変更

## 受け入れ条件

- forced hflip で `anchor_raw_indices_view` が `project_trace_label_view(anchor_raw_indices, ...)` と一致する。
- forced `space.factor_h` で `anchor_raw_indices_view` が `project_trace_label_view(anchor_raw_indices, ...)` と一致する。
- どちらの test でも `trace_valid == False` の row は `-1`。
- 既存の `fb_idx_coarse`, `offsets`, `coarse_usable_label_view` 関連 test が壊れない。

## 確認コマンド

まず追加 test だけを確認する。

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse.py \
  -k 'anchor_raw_indices_view or projects_space_labels_and_offsets' \
  -q
```

最後に coarse test 全体を通す。

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse.py -q
```

## 実装上の注意

`project_trace_label_view()` は integer label 用の helper で、0 を invalid 扱いしない。raw trace index には 0 が有効値としてあり得るため、この helper を使うのが正しい。

`project_fb_idx_view()` を raw trace index に使ってはいけない。`project_fb_idx_view()` は 0 を無効扱いし、時間方向 projection も適用するため、raw trace identity が壊れる。
